website/docs/configuration.md

```diff
@@ -3,77 +3,100 @@ id: configuration
 title: Configuration
 ---
 
-Docusaurus asks for only a few fields to configure your site.
+<!-- Goal: To explain the intention and best practices for configurations -->
 
-## `docusaurus.config.js`
+Docusaurus has a unique take on configurations. We encourage you to congregate information of your site into one place. We will guard the fields of this file, and facilitate making this data object accessible across your site.
 
-When you create your site using `docusaurus init`, the `docusaurus.config.js` file is initialized alongside the template.
+Keeping a well-maintained `docusaurus.config.js` helps you, your collaborators, and your open source contributors be able to focus on docs while having certain fields easy to customize.
 
-### Configurable fields
+## What goes into `docusaurus.config.js`?
 
-`docusaurus.config.js` contains configurable fields for:
+You should not have to write your `docusaurus.config.js` from scratch even if you are developing your site. All templates should come with a `docusaurus.config.js` at root that includes the necessary fields for the initial site.
 
-- Meta data of your site
-  - `title: string`
-  - `tagline: string`
-  - `favicon: string` relative url to your site's logo
-- Deployment fields for GitHub Pages, see [deployment configurations](/docs/deployment#docusaurusconfigjs-settings)
-  <!-- TODO: explain that theme configurations will be consumed by the theme, and link to theme doc -->
-- Theme configurations, see [theme configurations](#)
-  <!-- TODO: explain that preset configurations will be used to define presets of the site, and link to doc -->
-- Preset configurations, see [preset configurations](#)
+However, it can be helpful if you have a high-level understanding of how the configurations are designed and implemented.
 
-```js
-module.exports = {
-  // Site meta
-  title: 'Docusaurus',
-  tagline: '⚡️ Painless static site generator',
-  favicon: 'img/docusaurus.ico',
-
-  // Deployment fields
-  organizationName: 'facebook',
-  projectName: 'docusaurus',
-  baseUrl: '/',
-  url: 'https://v2.docusaurus.io',
-
-  themeConfig: {
-    // theme specific configurations
-  },
-  presets: [
-    // presets configurations
-  ],
-};
-```
+For reference to each of the configurable fields, you may refer to the API reference of [docusaurus.config.js](/docs/docusaurus.config.js).
+
+The configurations can be categorized into:
+
+- [Site meta](#site-meta)
+- [Deployment configurations](#deployment-configurations)
+- [Theme configurations, plugins, and presets](#theme-plugins-and-presets-configurations)
+- [Custom configurations](#custom-configurations)
+
+### Site meta
+
+Site meta contains the essential meta information such as titles and `favicon`.
+
+They are used by your site app in a number of places such as your site's title and headings, browser tab icon, and SEO.
+
+- [title](/docs/docusaurus.config.js#title)
```

Comment:
use filename linking like v1. Otherwise this is kinda hardcoded

https://docusaurus.io/docs/en/doc-markdown#linking-other-documents

Reply:
this make it works on github too